### ISSUE TITLE
Alter removes a reference variable, breaking the code after

### ISSUE THREAD

**Issue by williamdes** [vimeo/psalm#6114]
https://psalm.dev/r/7e011cd6cc

Source: https://github.com/phpmyadmin/phpmyadmin/blob/1835ebb4fccc20aa06868be88ab8bcaa9c395ccf/libraries/classes/Display/Results.php#L333

It removes the assignment (`$relDb = &`), but if you do that the code below will not work.

Same [here](https://github.com/phpmyadmin/phpmyadmin/blob/1835ebb4fccc20aa06868be88ab8bcaa9c395ccf/libraries/classes/Plugins/Schema/Pdf/Pdf.php#L401): 
```diff
diff --git a/libraries/classes/Plugins/Schema/Pdf/Pdf.php b/libraries/classes/Plugins/Schema/Pdf/Pdf.php
index 6e3dae46b2..90f956648e 100644
--- a/libraries/classes/Plugins/Schema/Pdf/Pdf.php
+++ b/libraries/classes/Plugins/Schema/Pdf/Pdf.php
@@ -398,7 +398,7 @@ class Pdf extends PdfLib
      */
     public function numLines($w, $txt)
     {
-        $cw = &$this->CurrentFont['cw'];
+        $this->CurrentFont['cw'];
         if ($w == 0) {
             $w = $this->w - $this->rMargin - $this->x;
         }
```

**Comment by orklah** [comment]
Psalm already prevent emitting UnusedVariable on references. After https://github.com/vimeo/psalm/pull/6129, it will also do that for globals. I think it'd also need to do it for variables that are references of others.

**Comment by AndrolGenhald** [comment]
I believe this is now fixed, likely by #7339. I tested by:

 - Checking out that phpmyadmin commit
 - `composer update`
 - `vendor/bin/psalm --alter --issues=UnusedVariable libraries/classes/Display/Results.php`

And `$relDb = &` was removed, then:

 - `git checkout .`
 - `composer require --dev vimeo/psalm="dev-master"`
 - Fix psalm.xml by removing `totallyTyped="false"`
 - `vendor/bin/psalm --alter --issues=UnusedVariable libraries/classes/Display/Results.php`

And the line is not changed.
